src/components/ebay-textbox/index.js

```diff
@@ -28,14 +37,22 @@ function getInitialState(input) {
         iconPostfix,
         tag: input.fluid ? 'div' : 'span',
         textboxTag: Boolean(input.multiline) ? 'textarea' : 'input',
-        invalid: String(Boolean(input.invalid))
+        invalid: String(Boolean(input.invalid)),
```

Comment:
Some of these values are evaluated here in the return, while others are above. Should we consolidate them all to above now, to make things just a little more readable and easier to follow?

Reply:
Are you thinking something like this?
```
    function getInitialState(input) {
        const { floatingLabel, icon, style } = input;
        const iconTag = input.iconTag && input.iconTag.renderBody;
        const invalid = String(Boolean(input.invalid));
        const tag = input.fluid ? 'div' : 'span';
        const textboxTag = Boolean(input.multiline) ? 'textarea' : 'input';
        const classes = ['textbox__control'];
        const displayIcon = Boolean(input.icon && !input.multiline && input.iconTag);
        const iconPostfix = input.iconPosition === 'postfix';
        const iconPrefix = input.iconPosition === 'prefix' || !iconPostfix;
        const rootClass = ['textbox', input.class];
        const labelClasses = ['floating-label__label'];
        const htmlAttributes = processHtmlAttributes(input);
        if (input.fluid) {
            classes.push('textbox__control--fluid');
        }
        if (input.floatingLabel) {
            classes.push('textbox__control--underline');
        }
        if (displayIcon && iconPostfix) {
            rootClass.push('textbox--icon-end');
        }
        if (htmlAttributes.disabled) {
            labelClasses.push('floating-label__label--disabled');
        }

        return {
            htmlAttributes,
            rootClass,
            style,
            classes,
            icon,
            iconTag,
            displayIcon,
            iconPrefix,
            iconPostfix,
            tag,
            textboxTag,
            invalid,
            floatingLabel,
            labelClasses
        };
    }
```
Wasn't sure if changing lines around so much (consolidating consts and if statements) constitutes a broken window or not, but let me know what you think. Could destructure even more of the `input.x` make it look cleaner by dropping the `input.` part, though maybe at cost of ease to understand now that the link to input is removed.